exchanges/bithumb/bithumb_websocket.go

```diff
@@ -41,24 +41,31 @@ func (b *Bithumb) WsConnect() error {
 			b.Name,
 			err)
 	}
+
+	b.Websocket.Wg.Add(1)
 	go b.wsReadData()
+
 	b.setupOrderbookManager()
 	return nil
 }
 
 // wsReadData receives and passes on websocket messages for processing
 func (b *Bithumb) wsReadData() {
-	b.Websocket.Wg.Add(1)
 	defer b.Websocket.Wg.Done()
 
 	for {
-		resp := b.Websocket.Conn.ReadMessage()
-		if resp.Raw == nil {
+		select {
+		case <-b.Websocket.ShutdownC:
```

Comment:
I don't believe this shutdown listener is needed in this specific instance due to your other improvements. 
If the ws is disconnected: 
- readmessage gets hit
- It will return a disconnected error
- Readmessage will attempt to relay that
  - If it cannot, you've added the default case to log a warn
- It will then return the `Response{}` with a nil `.Raw`, causing the function to return and `Wg.Done()` to be hit.

Reply:
For some reason, that was not the case. And I am unable to find out why. But this fixed the issue.

Reply:
Thankfully what I outlined above worked for me. I won't get you to remove it, but it would have meant all `wsReadData()` implementations would need a shutdown receiver